lib/engine/action/protect.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Action
+    class Protect < Base
```

Comment:
Would prefer this to be move explicit, what are you protecting? 'PriceProtect'

Reply:
Good point. It was also a bit inconsistent, because I did that in some places